Documentation/installation-with-spack.rst

```diff
@@ -0,0 +1,391 @@
+  module load <compiler>
+
+
+After that, you can compiler SiesSol using either CMake or 
+:ref:`Scons <compiling-seissol>`.
+
+
+SeisSol-Utils
+-------------
+
+By default, the scrip installs:
+
+- pumgen (without a Simmetrix support)
+- gmsh (without a GPU support)
+- gmsh2gambit
+- cube_c
+- rconv
+- SiesSol Cookbook, which contains some examples to run
+
+As in case of *seissol-env*, you need **scons** and, therefore, **python3** for 
+compiling. However, installation of these packages is optional to save 
+the set-up time.
+
+
+Options
+~~~~~~~
+
+- *benchmarks* [on, **default=off**] - installs SeisSol benchmarks. Make sure that you have access to the SeisSol LRZ-gitlab account.
+- *building_tools* [on, **default=off**] - installs scons and as a result python and pip
+- *gmsh_gui* [on, **default=off**] - enables gui support for gmsh
+- *paraview* [on, **default=off**] - installs Paraview for visualization
+
+Examples
+~~~~~~~~
+
+.. code-block:: bash
+
+  # 1. essential packages compiled with gcc compiler suite
+  spack install seissol-utils %gcc@8.3.0
+
+  # 2. with benchmarks and gmsh gmsh GUI
+  spack install seissol-utils+gmsh_gui+benchmarks %gcc@8.3.0
+
+  # 3. with gmsh GUI, paraview and scons
+  spack install seissol-utils+gmsh_gui+paraview+building_tools %gcc@8.3.0
+
+  # 4. essential packages with simmetrix support for pumgen
+  spack install seissol-utils ^pumgen+simmetrix_support %gcc@8.3.0 
+
+Usage
+~~~~~
+
+.. code-block:: bash
+
+  module load seissol-utils-develop-<compiler>-<hash>
+
+  # to access the Cookbook
+  cd $COOKBOOK
+
+  # to access the Benchmakrs
+  cd $BENCHMAKRS
+
+
+
+Tips and Tricks
+---------------
+
+.. _spack_known_issues:
+
+1. Spack builds the entire dependency graph before compaling and installing. 
+The graph includes all libs and packages which are necessary to build your 
+application, including packages like: *tar, gzip, zlib,  autoconf, 
+cmake, automake, pkgconf, m4, ncurses, etc*. Packages like these do not 
+affect performance of your application but help Spack to install it. 
+Therefore, it is not necessary to install them again and again. You can 
+install such  packages only once and mark them as Default 
+`(External) <https://spack-tutorial.readthedocs.io/en/latest/tutorial_configuration.html#external-packages>`_.
+and Non-Buildable. It can speed-up installation of SeisSol-Env and SeisSol-utils 
+considerably. You will need to modify and edit **~/.spack/packages.yaml** file.
+
+
+Known Issues
+------------
+
+1. Spack is a really live project with dozens of commits per day. It is 
+diffucult for us to keep the same pace with Spack. A new version of Spack
+may not work because of new added features what we may not be aware of. 
+Therefore, it may be necessary to use an older version of Spack. You
+can simply do it by moving the HEAD of your locally installed Spack
+repository to an old commit:
+
+.. code-block:: bash
+
+  cd $SPACK_ROOT
+  git checkout <a previous SPACK commit>
```

Comment:
Maybe provide todays spack commit hash?

Reply:
Honestly, I am not sure about it. I cannot check all possible combinations with respect to all possible installation options. Therefore, I am not sure that today's snippet of Spack is a perfect reference

Reply:
Okay, I see there are just to many options to try all of them. But maybe we can add something like: On SuperMUC commit xxxxxx works fine for the combinations described above.

Reply:
It is a good idea. Additionally, we can provide package.yaml file with not-buildable packages (an use it as an exaple) and a specify a particular commit for Spack. Let me see whether I can do it for SuperMUC because the admins did whatever they could to prevent compiling on production nodes but maybe i will manage to find a way around